### PR TITLE
client: do not send previous node data (optional)

### PR DESCRIPTION
**Pull request by gyuho**
- Do not send back node data when specified
- remove node and prevNode when noDataOnSuccess is set

Closing
- https://github.com/coreos/etcd/pull/3907
- https://github.com/coreos/etcd/issues/3363

@fraenkel provided test cases as well.

I just resolved merge conflicts.

/cc @xiang90

### CODE REVIEW

File: client/keys.go [hunk 1]
```diff
@@ -191,6 +191,10 @@ type SetOptions struct {
 
 	// Dir specifies whether or not this Node should be created as a directory.
 	Dir bool
+
+	// NoDataOnSuccess specifies whether the response contains the current value of the Node.
+	// If set, the response will only contain the current value when the request fails.
+	NoDataOnSuccess bool
```
Comment (xiang90) [inline client/keys.go]: NoValueOnSuccess?

**Comment by xiang90** [comment]
lgtm otherwise.

**Comment by gyuho** [comment]
Ok will change that. (Also need to fix test failures)

**Comment by xiang90** [comment]
Add the new option with an example to the v2 doc here https://github.com/coreos/etcd/blob/master/Documentation/v2/api.md#atomic-compare-and-swap?

**Comment by gyuho** [comment]
Added an example to that doc. Merging after test passes. Thanks.

File: Documentation/v2/api.md [hunk 2]
```diff
@@ -559,6 +559,25 @@ Let's create a key-value pair first: `foo=one`.
...
+    "node":{
+        "key":"/foo",
+        "value":"one",
+        "modifiedIndex":4,
+        "createdIndex":4
+    }
+}
+```
+
+Specifying `noValueOnSuccess` option skips returning this data
```
Comment (xiang90) [inline Documentation/v2/api.md]: skips returning the node as value.

**Comment by xiang90** [comment]
Thanks a lot!

**Pull request merged**